spec/controllers/submissions_controller_spec.rb

```diff
@@ -0,0 +1,633 @@
+                'Content of Shapes.java')
+        txn.add(File.join(@assignment.repository_folder, 'TestShapes.java'),
+                'Content of TestShapes.java')
+        repo.commit(txn)
+        
+        # revision 2
+        revision = repo.get_latest_revision
+        old_files = revision.files_at_path(@assignment.repository_folder)
+        old_file_1 = old_files['Shapes.java']
+        old_file_2 = old_files['TestShapes.java']
+
+        @file_1 = fixture_file_upload(File.join('/files', 'Shapes.java'),
+                                      'text/java')
+        @file_2 = fixture_file_upload(File.join('/files', 'TestShapes.java'),
+                                      'text/java')
+
+        post_as @student,
+                :update_files,
+                assignment_id: @assignment.id,
+                new_files: [@file_1, @file_2],
+                file_revisions: { 'Shapes.java'     => old_file_1.from_revision,
+                                  'TestShapes.java' =>
+                                      old_file_2.from_revision }
+      end
+      is_expected.to respond_with(:success)
+
+      expect(assigns :assignment).to_not be_nil
+      expect(assigns :grouping).to_not be_nil
+      expect(assigns :path).to_not be_nil
+      expect(assigns :revision).to_not be_nil
+      expect(assigns :files).to_not be_nil
+      expect(assigns :missing_assignment_files).to_not be_nil
+
+      @grouping.group.access_repo do |repo|
+        revision = repo.get_latest_revision
+        files = revision.files_at_path(@assignment.repository_folder)
+        expect(files['Shapes.java']).to_not be_nil
+        expect(files['TestShapes.java']).to_not be_nil
+
+        # Test to make sure that the contents were successfully updated
+        @file_1.rewind
+        @file_2.rewind
+        file_1_new_contents = repo.download_as_string(files['Shapes.java'])
+        file_2_new_contents = repo.download_as_string(files['TestShapes.java'])
+
+        expect(@file_1.read).to eq(file_1_new_contents)
+        expect(@file_2.read).to eq(file_2_new_contents)
+      end
+
+    end
+
+    it 'should be able to delete files' do
+      expect(@student.has_accepted_grouping_for?(@assignment.id)).to be_truthy
+
+      @grouping.group.access_repo do |repo|
+        txn = repo.get_transaction('markus')
+        txn.add(File.join(@assignment.repository_folder,'Shapes.java'),
+                'Content of Shapes.java')
+        txn.add(File.join(@assignment.repository_folder, 'TestShapes.java'),
+                'Content of TestShapes.java')
+        repo.commit(txn)
+        revision = repo.get_latest_revision
+        old_files = revision.files_at_path(@assignment.repository_folder)
+        old_file_1 = old_files['Shapes.java']
+        old_file_2 = old_files['TestShapes.java']
+
+        post_as(@student, :update_files, {:assignment_id => @assignment.id,
+          :delete_files => ['Shapes.java'],
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.